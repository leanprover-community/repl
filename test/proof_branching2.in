{"cmd": "example (p q : Prop) : p ∧ q → q ∧ p := by sorry", "sorries": "individual"}

{"tactic": "intro h", "proofState": 0}

{"tactic": "have hp : p := h.left", "proofState": 1}

{"tactic": "have hq : q := h.right", "proofState": 2}

{"tactic": "apply And.intro", "proofState": 3}

{"tactic": "exact hq", "proofState": 4}

{"tactic": "exact hp", "proofState": 5}

