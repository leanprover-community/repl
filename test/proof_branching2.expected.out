{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 43},
   "parentDecl": "_example",
   "goals": ["p q : Prop\n⊢ p ∧ q → q ∧ p"],
   "endPos": {"line": 1, "column": 48}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 1,
 "goals": ["p q : Prop\nh : p ∧ q\n⊢ q ∧ p"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals": ["p q : Prop\nh : p ∧ q\nhp : p\n⊢ q ∧ p"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 3,
 "goals": ["p q : Prop\nh : p ∧ q\nhp : p\nhq : q\n⊢ q ∧ p"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 4,
 "goals":
 ["case left\np q : Prop\nh : p ∧ q\nhp : p\nhq : q\n⊢ q",
  "case right\np q : Prop\nh : p ∧ q\nhp : p\nhq : q\n⊢ p"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 5,
 "goals": ["case right\np q : Prop\nh : p ∧ q\nhp : p\nhq : q\n⊢ p"]}

{"proofStatus": "Completed", "proofState": 6, "goals": []}

